esphome/components/mqtt/mqtt_backend_esp32.h

```diff
@@ -42,9 +46,71 @@ struct Event {
         error_handle(*event.error_handle) {}
 };
 
+enum MqttQueueTypeT : uint8_t {
+  MQTT_QUEUE_TYPE_NONE = 0,
+  MQTT_QUEUE_TYPE_SUBSCRIBE,
+  MQTT_QUEUE_TYPE_UNSUBSCRIBE,
+  MQTT_QUEUE_TYPE_PUBLISH,
+};
+
+struct QueueElement {
+  char *topic;
+  char *payload;
+  uint16_t payload_len;  // MQTT max payload is 64KiB
+  uint8_t type : 2;
+  uint8_t qos : 2;  // QoS only needs values 0-2
+  uint8_t retain : 1;
+  uint8_t reserved : 3;  // Reserved for future use
+
+  QueueElement() : topic(nullptr), payload(nullptr), payload_len(0), qos(0), retain(0), reserved(0) {}
+
+  // Helper to set topic/payload (handles malloc)
+  bool set_data(const char *topic_str, const char *payload_data, size_t len) {
+    // Check payload size limit (MQTT max is 64KiB)
+    if (len > UINT16_MAX) {
+      return false;
+    }
+
+    topic = strdup(topic_str);
+    if (!topic)
+      return false;
+
+    if (payload_data && len) {
+      payload = (char *) malloc(len);  // NOLINT
+      if (!payload) {
+        free(topic);  // NOLINT
+        topic = nullptr;
+        return false;
+      }
+      memcpy(payload, payload_data, len);
+      payload_len = static_cast<uint16_t>(len);
+    } else {
+      payload = nullptr;
+      payload_len = 0;
+    }
+    return true;
+  }
+
+  // Helper to release (handles free)
+  void release() {
+    if (topic) {
+      free(topic);  // NOLINT
+      topic = nullptr;
+    }
+    if (payload) {
+      free(payload);  // NOLINT
+      payload = nullptr;
+    }
+    payload_len = 0;
+  }
+};
+
 class MQTTBackendESP32 final : public MQTTBackend {
  public:
   static const size_t MQTT_BUFFER_SIZE = 4096;
+  static const size_t TASK_STACK_SIZE = 4096;
```

Comment:
```suggestion
  static const size_t TASK_STACK_SIZE = 2048;
```

I think we can get away with 2048